packages/p2p/src/components/modal-manager/modals/my-ads-floating-rate-switch-modal/my-ads-floating-rate-switch-modal.tsx

```diff
@@ -38,10 +38,14 @@ const MyAdsFloatingRateSwitchModal = ({ onSwitch }: TMyAdsFloatingRateSwitchModa
                         className='my-ads-floating-rate-switch-modal__cancel-button'
                         secondary
                         type='button'
-                        onClick={() => toggleMyAdsRateSwitchModal(selected_ad_type, false)}
+                        onClick={() => toggleMyAdsRateSwitchModal(selected_ad_type, p2p_settings?.reached_target_date)}
                         large
                     >
-                        <Localize i18n_default_text='Cancel' />
+                        {p2p_settings?.reached_target_date ? (
```

Comment:
do we still need this condition?

Reply:
yes we still need to check for this, because it can be undefined and this needs to be shown to handle what is to be shown 👍 i had discussed this with dharshini earlier